src/spi.rs

```diff
@@ -961,29 +978,32 @@ where
     fn spi_transactions<'a>(
         &self,
         operations: impl Iterator<Item = Operation<'a, u8>> + 'a,
-    ) -> impl Iterator<Item = spi_transaction_t> + 'a {
-        enum OperationsIter<R, W, T, I> {
+    ) -> impl Iterator<Item = (spi_transaction_t, u32)> + 'a {
```

Comment:
To wrap the extra info externally is one way to archive the goal here. At least it is the only approach if we want to still use the flat_map into native spi_transatcion approach ongoing. Though with the Delay operator needed now it all feels a bit hacked together ...

Reply:
Well I don't want to rewrite the whole driver just because of the delay operation. Moreover, I'm not sure that's a big hack. In a way, (spi_transaction_t, u32) is an "either" operation. Either a transaction, or a delay. We can model it with a dedicated `Either` type too, or abuse `Result` for that.

Reply:
In any case, we now have an `SpiOperation` enum, that serves the role of "either" an `spi_transaction_t` or a delay.